scbctl/cmd/cascading_test.go

```diff
@@ -0,0 +1,222 @@
+// SPDX-FileCopyrightText: the secureCodeBox authors
+//
+// SPDX-License-Identifier: Apache-2.0
+package cmd
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/ddddddO/gtree"
+	cascadingv1 "github.com/secureCodeBox/secureCodeBox/operator/apis/cascading/v1"
+	v1 "github.com/secureCodeBox/secureCodeBox/operator/apis/execution/v1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MockCascadeClientProvider struct {
+	client.Client
+	namespace string
+	err       error
+}
+
+func (m *MockCascadeClientProvider) GetClient(_ *genericclioptions.ConfigFlags) (client.Client, string, error) {
+	return m.Client, m.namespace, m.err
+}
+
+type testcases struct {
+	name           string
+	args           []string
+	expectedError  error
+	expectedOutput string
+	initialScans   []v1.Scan
+	initialRules   []cascadingv1.CascadingRule
+}
+
+func TestBuildTree(t *testing.T) {
+	tests := []struct {
+		name     string
+		scans    []v1.Scan
+		expected string
+	}{
+		{
+			name: "Single scan",
+			scans: []v1.Scan{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "scan1",
+					},
+				},
+			},
+			expected: "Scans\n└── scan1\n",
+		},
```

Comment:
You can use golang multi-line strings here to make the expected output a bit more readable :)
```suggestion
			expected: `Scans
└── scan1
`,
```